recipe_scrapers/projectgezond.py

```diff
@@ -11,82 +10,32 @@ def host(cls):
     def author(self):
         return "Project Gezond"
 
-    def title(self):
-        return self.soup.find("h1", {"class": "entry-title"}).text
-
-    def category(self):
-        return ", ".join(
-            [
-                element.text
-                for element in self.soup.find(
-                    "span", {"class": "meta-category"}
-                ).find_all(
-                    "a", {"class": lambda x: x is not None and x.startswith("category")}
-                )
-            ]
-        )
-
-    def total_time(self):
-        time_element = self.soup.find("em", string="Bereidingstijd:").parent
-        return "".join(
-            [
-                element.text
-                for element in time_element.children
-                if element.text != "Bereidingstijd:"
-            ]
-        ).strip()
-
-    def yields(self):
-        # Match everything in the h2 with 'Dit heb je nodig'
-        # The text inside the parentheses contains the yield for the ingredients that are listed
-        return re.search(
-            r"Dit heb je nodig \(([^)]+)",
-            self.soup.find(
-                "h2", string=lambda x: x.startswith("Dit heb je nodig")
-            ).text,
-        ).group(1)
-
-    def ingredients(self):
-        ingredients_table = self.soup.find(
-            "h2", string=lambda x: x.startswith("Dit heb je nodig")
-        ).next_sibling.next_sibling
-        ingredients = [
-            ingredient.text
-            for ingredient in ingredients_table
-            if ingredient.text.strip()
-        ]
-        return ingredients
-
-    def instructions(self):
-        instructions_table = self.soup.find(
-            "h2", string=lambda x: x.startswith("Zo maak je het")
-        ).next_sibling.next_sibling.next_sibling.next_sibling
-        instructions = [
-            instruction.text
-            for instruction in instructions_table
-            if instruction.text.strip()
-        ]
-        return "\n".join(instructions).strip()
-
     def ratings(self):
-        # Ratings do not exist on this site
-        return None
+        raise FieldNotProvidedByWebsiteException(return_value=None)
 
     def cuisine(self):
-        # Not listed on site
-        return None
-
-    def description(self):
-        # Get the recipe's content start. The recipe will start with the description until
-        # we reach the instructions.
-        content_start = self.soup.find("div", {"class", "entry-content"})
+        raise FieldNotProvidedByWebsiteException(return_value=None)
```

Comment:
If we begin following [semantic versioning](https://semver.org/) for our releases, should we consider this a non-backwards-compatible change?

Reply:
Yes, I think so. The new exception would need to be handled explicitly, which could break existing setups depending on how they currently manage missing fields.